### PR TITLE
core changes required for authentication helper add on v1.1.0

### PR DESCRIPTION
**Pull request by KajanM**
- [x] required changes for guided configuration
- [x] required changes for authentication status scanning
- [x] required changes for automated configuration

Above changes are parallelly applied and manually tested in authenticationhelper add-on v2.
You can check them out at
- https://github.com/KajanM/zap-extensions/tree/authenticationhelper-addon-1.1.0
- https://github.com/KajanM/zap-extensions/tree/authenticationhelper-v2-with-updated-zap

Some changes are not used by the add-on, however, they are added assuming they may help other developers.

### CODE REVIEW

File: src/org/parosproxy/paros/network/HttpSender.java [hunk 1]
```diff
@@ -131,6 +131,7 @@
 	public static final int FORCED_BROWSE_INITIATOR = 11;
 	public static final int TOKEN_GENERATOR_INITIATOR = 12;
 	public static final int WEB_SOCKET_INITIATOR = 13;
+	public static final int AUTHENTICATION_HELPER_INITIATOR = 14;
```
Comment (thc202) [inline src/org/parosproxy/paros/network/HttpSender.java]: A ZAP comment should be added (after the license header).

File: src/org/zaproxy/zap/view/StandardFieldsDialog.java [hunk 2]
```diff
@@ -1159,6 +1159,43 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
 		this.incTabOffset(tabIndex);
 	}
 	
+	/**
+	 * Add a 'node select' field which provides a button for showing a Node Select Dialog and a 
+	 * non editable field for showing the node selected
+	 * 
+	 * @since 2.8.0
```
Comment (thc202) [inline src/org/zaproxy/zap/view/StandardFieldsDialog.java]: We usually add the exact version when in a release branch, until then it's just a `TODO add version`.

File: src/org/zaproxy/zap/view/StandardFieldsDialog.java [hunk 3]
```diff
@@ -1359,6 +1396,81 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
...
+		this.addField(this.tabPanels.get(tabIndex), this.tabOffsets.get(tabIndex), componentLabel, component, component, 0.0D);
+		this.incTabOffset(tabIndex);
+	}
+	
+	/**
+	 * Add a custom {@code Component} to {@code StandardFieldsDialog} without any label
+	 * @param component
+	 * @since 2.8.0
+	 */
+	public void addCustomComponent(Component component) {
```
Comment (thc202) [inline src/org/zaproxy/zap/view/StandardFieldsDialog.java]: Could you move this to after the corresponding tabbed method instead?

**Comment by KajanM** [comment]
comments addressed, thanks.

File: src/org/zaproxy/zap/view/StandardFieldsDialog.java [hunk 4]
```diff
@@ -1159,6 +1159,43 @@ public void actionPerformed(java.awt.event.ActionEvent e) {
 		this.incTabOffset(tabIndex);
 	}
 	
+	/**
+	 * Add a 'node select' field which provides a button for showing a Node Select Dialog and a 
+	 * non editable field for showing the node selected
+	 * 
+	 * TODO: add version
```
Comment (thc202) [inline src/org/zaproxy/zap/view/StandardFieldsDialog.java]: Sorry, I was not clear, it should still have the tag, e.g. `@since TODO add version`.

**Comment by thc202** [comment]
btw, the 3rd task would be handled in another PR, right?

File: src/org/zaproxy/zap/view/StandardFieldsDialog.java (hunk 4 above)
Comment (KajanM) [reply src/org/zaproxy/zap/view/StandardFieldsDialog.java]: no problem

**Comment by KajanM** [comment]
I planned to handle the 3rd task in this PR itself, if a separate PR is more approprieate let me know :)

**Comment by thc202** [comment]
Yeah, it's fine in this PR.

**Comment by KajanM** [comment]
updated description with 

> Above changes are parallelly applied and manually tested in authenticationhelper add-on v2.
You can check them out at
> - https://github.com/KajanM/zap-extensions/tree/authenticationhelper-addon-v2
> - https://github.com/KajanM/zap-extensions/tree/authenticationhelper-v2-with-updated-zap
> Some changes are not used by the add-on, however, they are added assuming they may help other developers.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**